config/file_config.go

```diff
@@ -75,19 +76,18 @@ func parseNodesAndPeers(value interface{}) (nodes map[string]FederationNode, pee
 	return nodes, peers, err
 }
 
-func populateConfig(cfg mutableNodeConfig, data map[string]interface{}) (err error) {
+func populateConfig(cfg mutableNodeConfig, data map[string]interface{}) (error) {
 	for key, value := range data {
 		var duration time.Duration
 		var numericValue uint32
 		var publicKey primitives.Ed25519PublicKey
+		var err error
 
 		switch value.(type) {
 		case float64:
 			numericValue, err = parseUint32(value.(float64))
 		case string:
-			if parsedDuration, err := time.ParseDuration(value.(string)); err == nil {
-				duration = parsedDuration
-			}
+			duration, err = time.ParseDuration(value.(string))
```

Comment:
didn't you change behavior here? what if the parse failed?

Reply:
i looked and if parse fails the duration is 0.

Reply:
ok, but in general does it make sense to have a zero duration as default?